### PR TITLE
extract clusterOpts for readability

### PR DESCRIPTION
**Pull request by utako**

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -235,11 +268,14 @@ func (configgen *ConfigGeneratorImpl) buildOutboundSniDnatClusters(env *model.En
 					}
 					subsetCluster := buildDefaultCluster(env, subsetClusterName, discoveryType, lbEndpoints, model.TrafficDirectionOutbound, proxy)
 					subsetCluster.TlsContext = nil
-					applyTrafficPolicy(env, subsetCluster, destinationRule.TrafficPolicy, port, nil, "",
-						SniDnatClusterMode, model.TrafficDirectionOutbound, proxy)
-					applyTrafficPolicy(env, subsetCluster, subset.TrafficPolicy, port, nil, "",
-						SniDnatClusterMode, model.TrafficDirectionOutbound, proxy)
+
+					opts.cluster = subsetCluster
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: I think this makes it actually harder to follow what is going on, now I need to keep track of how opts was initialized, and find all the places fields changed to understand what is being passed. Before it was very explicit

Comment (utako) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: yeah, I can go either way. I still like extracting the variables into the buildClusterOpts object. i've modified the PR to keep things explicit.

**Comment by howardjohn** [comment]
Tests seem stuck?
/test all

**Comment by howardjohn** [comment]
/test integ-pilot-local-presubmit-tests-master
Infra flake, it says it passed but reported failure? https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/14345/integ-pilot-local-presubmit-tests-master/939

**Pull request merged**